src/firmware/app/control/trajectory_planner.h

```diff
@@ -72,232 +71,369 @@ typedef enum TrajectoryPlannerGenerationStatus
+ * the trajectory. In seconds.
+ *
+ * @return A status indicating whether or not generation was successful
+ */
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_interpolateConstantPeriodPositionTrajectory_2(
+    PositionTrajectory_t* constant_period_trajectory,
+    PositionTrajectory_t* variable_period_trajectory, unsigned int* num_elements,
+    float interpolation_period);
+
+/**
+ * Function generates a constant interpolation period trajectory from the path parameters
+ * specified in the constant parameterization trajectory. The function creates a constant
+ * parameterization trajectory then uses linear interpolation to calculate the equivalent
+ * constant interpolation period trajectory
+ *
+ * NOTE: Constant interpolation period means that the duration of time between each point
+ * on the trajectory is CONSTANT. This is different from a constant parameterization
+ * trajectory which has segments defined at constant polynomial evaluation intervals.
+ * Constant arc-length trajectories (not used here) are trajectories where the distance
+ * travelled in each segment in constant.
+ *
+ * @param constant_period_trajectory [out] A constant interpolation period trajectory
+ * generated by the specified path parameters.
+ *
+ * @param path_parameters [in/out] The path parameters that define the trajectory. The
+ * input path_parameters.num_elements defines the number of constant-parameterization
+ * elements, and the returned path_parameters.num_elements defines the number of
+ * constant-period elements in the trajectory,
+ *
+ * @param interpolation_period  [in] The duration of time between successive trajectory
+ * elements. In seconds.
+ *
+ * @return A status indicating whether or not generation was successful
  */
-void static app_trajectory_planner_generatePositionTrajectoryTimeProfile(
-    PositionTrajectoryElement_t* traj_elements, const float num_segments,
-    const float arc_segment_length, float* velocity_profile);
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_generateConstantInterpolationPeriodPositionTrajectory_2(
+    PositionTrajectory_t* constant_period_trajectory,
+    FirmwareRobotPathParameters_t* path_parameters, float interpolation_period);
 
-/***
- *  This function takes in a forwards trajectory annd modifies it in place to become a
- * backwards trajectory NOTE: This function exists to avoid issues outlined in #1322
+/**
+ * Function generates a velocity trajectory composed of X/Y/angular velocities and their
+ * time dependence.
+ *
+ * @param position_trajectory [in] A completely defined position trajectory.
  *
- *  TODO: Remove when #1322 is merged.
+ * @param velocity_trajectory [out] A completely defined velocity trajectory that has
+ * equally time-spaced elements.
  *
- * @param forwards_trajectory This is the trajectory that will be modified in place to
- * become a reverse trajectory
+ * @param num_elements [in] The number of elements in the trajectory.
```

Comment:
is the output `velocity_trajectory` guaranteed to have the same number of elements?

Reply:
Yep, added a comment clarifying 👍